internal/fetchclient/fetchclient.go

```diff
@@ -115,6 +118,52 @@ func (c *Client) fetchDartFlutter(ctx context.Context, name string) (string, err
 	return data.Latest.Version, nil
 }
 
+func (c *Client) fetchCrate(ctx context.Context, name string) (string, error) {
+	request, err := http.NewRequestWithContext(
+		ctx,
+		http.MethodGet,
+		fmt.Sprintf("%s/crates/%s", cratesURL, strings.TrimPrefix(name, "/")),
+		nil,
+	)
+	if err != nil {
+		return "", err
+	}
+	// See https://github.com/bufbuild/plugins/issues/252 for more information.
+	// We must be careful with this API and respect the crawling policy.
+	request.Header.Set("User-Agent", "bufbuild (github.com/bufbuild/plugins)")
+	response, err := c.httpClient.Do(request)
+	if err != nil {
+		return "", err
+	}
+	defer response.Body.Close()
+
+	var data struct {
```

Comment:
We might be able to just use `newest_version` or the `max_stable_version` instead of looping through all the versions.

Reply:
So far we only allow semver (without prerelease IIRC) so looping might be the best option.

Reply:
Ye, it's good to know these options are available. And there is also `:crate_id/versions` available.

https://github.com/rust-lang/crates.io/blob/8be79196dbcd7fb1388cf016ab128da11c484ed6/src/router.rs#L76-L79